packages/core/cli.js

```diff
@@ -56,7 +56,11 @@ if (userWantsGeneralHelp) {
 
 command
   .run(inputArguments, options)
-  .then(returnStatus => process.exit(returnStatus))
+  .then(returnStatus => {
+    typeof returnStatus !== undefined
```

Comment:
I'm assuming you mean `typeof returnStatus !== "undefined"`?  If so, it's simpler to just write `returnStatus !== undefined`.  (But either way, `typeof returnStatus` will never equal `undefined`!)

Btw, is this code implicitly assuming that `returnStatus` will always either be either `undefined` or an integer?  If so you might want to add a comment to that effect.

Reply:
Yeah sorry I messed this up. So maybe we should eventually change Truffle's architecture here. Currently each command's `run` function does not really have a standardized system for returning values to this context here. Right now it just completes, sometimes returns something, and throws if there is an error.

Would it make sense to devise some sort of a system where each `run` function never throws and instead returns some sort of object describing what happened. Perhaps it would have fields that would describe whether it was successful, a message if there was an error, miscellaneous data related to any given error, etc. What do you think of that?

Reply:
Huh, interesting idea, but I have no idea!  Seems like a thing for later regardless...